engine/orchestration/src/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java

```diff
@@ -1353,6 +1363,20 @@ public void prepare(VirtualMachineProfile vm, DeployDestination dest) throws Sto
             disk.setDetails(getDetails(volumeInfo, dataStore));
 
             vm.addDisk(disk);
+
+            // If hypervisor is vSphere, check for clone type setting.
+            if (vm.getHypervisorType().equals(HypervisorType.VMware)) {
```

Comment:
Does this need to be specific to VMWare? I can see a use case where this can be useful for XenServer as well. IIRC for XenServer, the volumes are `linked`.